crates/concrete_ast/src/expressions.rs

```diff
@@ -18,8 +18,8 @@ pub enum Expression {
 pub enum ValueExpr {
     ConstBool(bool),
     ConstChar(char),
-    ConstInt(u64),
-    ConstFloat(()),
+    ConstInt(u128),
+    ConstFloat(String),
```

Comment:
why float?

Reply:
due to salsa requirements

Reply:
we should using something like the OrderedFloat crate https://docs.rs/ordered-float/latest/ordered_float/struct.OrderedFloat.html

Reply:
I think making `f32` and `f64` implement `Eq` is wrong since or confusing. An example:
```rs
    ConstFloat(1.23) == ConstFloat(1.23) // Depends on the parsed binary value but should be the same assuming a deterministic float parser.
    ConstFloat(1.23) == ConstFloat(NaN) // Is never true, which is what we expected.
    ConstFloat(NaN) == ConstFloat(NaN) // Is never true, but we may want it to be true when comparing AST nodes.
```

What about having an enum like this? Or something similar, not necessarily this representation.
```rs
#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
enum FloatLiteral {
    Value {
        sign: Sign,
        integer: BigUint,
        decimal: BigUint,
    },
    NotANumber,
}
```